### PR TITLE
Fixing minor revision numbers read from old format fit files

### PR DESCRIPTION
**Pull request by egthomas**
This pull request addresses #203 by changing `fitopen.c` to correctly read the FITACF minor revision number from an old-format fit file.  I've also gone ahead and changed the FITACF version number in `fitacfversion.h` from 5.0 to 2.5 (this has also been done a long time ago in other branches which are still pending approval).  To test this pull request, on the develop branch follow these steps:

1. Create a `.fit` file from a `.dat` file, eg: `make_fit -old 1994123114k.dat 1994123114k.fit`
2. Create a `.fitacf` file from the `.fit` file:  `fittofitacf 1994123114k.fit > 1994123114k.fitacf`
3. Dump the contents of the `.fitacf` file:  `dmapdump 1994123114k.fitacf > fitacf.dump`
4. Check the `fitacf.revision.minor` value within `fitacf.dump` - it will be 0 instead of 5

Now switch to this branch and repeat the above steps.  The `fitacf.revision.minor` value will now contain the correct value (5) instead of 0.  You can further modify the version numbers in `fitacfversion.h` to test for robustness.

### CODE REVIEW

File: codebase/superdarn/src.lib/tk/oldfit.1.25/src/fitopen.c [hunk 1]
```diff
@@ -112,7 +112,7 @@ struct OldFitFp *OldFitOpenFd(int fitfp,int inxfp) {
            ptr->major_rev=atoi(tmpbuf+i-j);
            for (j=0;(ptr->header[i+1+j] !=' ') && 
                (ptr->header[i+1+j] !=0);i++) tmpbuf[j]=ptr->header[i+1+j]; 
-           tmpbuf[j]=0;
+           /*tmpbuf[j]=0;*/     /* Removing this so the minor_rev is correct */
```
Comment (mts299) [inline codebase/superdarn/src.lib/tk/oldfit.1.25/src/fitopen.c]: Are you going to delete this line before merging so we don't have commented code?

**Comment by mts299** [comment]
Tested this and it seems to work. 
Test 1. 
Develop branch
```Bash
$make_fit make_fit 20090320.1601.00.pgr.rawacf > 20090320.1601.00.pgr.fitacf 
$dmapdump 20090320.1601.00.pgr.fitacf | grep 'fitacf.revision.minor" = ' | head -n 1
$	int	"fitacf.revision.minor" = 0
```
fix_old_fitacf_minor_rev branch
```
$make_fit make_fit 20090320.1601.00.pgr.rawacf > 20090320.1601.00.pgr.fitacf 
$dmapdump 20090320.1601.00.pgr.fitacf | grep 'fitacf.revision.minor" = ' | head -n 1
$	int	"fitacf.revision.minor" = 5
```

Test 2
develop
```Bash
dattorawacf 2001102408w.dat > 2001102408w.rawacf
make_fit 2001102408w.rawacf > 2001102408w.fitacf
dmapdump 2001102408w.fitacf | grep 'fitacf.revision.minor" = ' | head -n 1
	int	"fitacf.revision.minor" = 0
```
fix_old_fitacf_minor_rev branch
```Bash
dattorawacf 2001102408w.dat > 2001102408w.rawacf
make_fit 2001102408w.rawacf > 2001102408w.fitacf
dmapdump 2001102408w.fitacf | grep 'fitacf.revision.minor" = ' | head -n 1
	int	"fitacf.revision.minor" = 5
```

Test 3 
develop
```Bash
make_fit -old 2001102408w.dat 2001102408w.fit
fittofitacf 2001102408w.fit > 2001102408w.fitacf 
dmapdump 2001102408w.fitacf | grep 'fitacf.revision.minor" = ' | head -n 1
	int	"fitacf.revision.minor" = 0
```
fix_old_fitacf_minor_rev branch
```Bash
make_fit -old 2001102408w.dat 2001102408w.fit
fittofitacf 2001102408w.fit > 2001102408w.fitacf 
dmapdump 2001102408w.fitacf | grep 'fitacf.revision.minor" = ' | head -n 1
	int	"fitacf.revision.minor" = 5
```

Test 4
develop 
```Bash
make_fit -fitacf-version 3.0 2001102408w.rawacf > 2001102408w.fitacf
dmapdump 2001102408w.fitacf | grep 'fitacf.revision.minor" = ' | head -n 1
	int	"fitacf.revision.minor" = 0
dmapdump 2001102408w.fitacf | grep 'fitacf.revision.major" = ' | head -n 1
	int	"fitacf.revision.major" = 3
```
fix_old_fitacf_minor_rev branch
```Bash
make_fit -fitacf-version 3.0 2001102408w.rawacf > 2001102408w.fitacf
dmapdump 2001102408w.fitacf | grep 'fitacf.revision.minor" = ' | head -n 1
	int	"fitacf.revision.minor" = 0
dmapdump 2001102408w.fitacf | grep 'fitacf.revision.major" = ' | head -n 1
	int	"fitacf.revision.major" = 3
```
Also, it fixed `fitacf.revision.major = 5` to `fitacf.revision.major = 2`

However, isn't fitacf default 3.0? Should these values not be 3.0?

**Comment by egthomas** [comment]
> However, isn't fitacf default 3.0? Should these values not be 3.0?

The default version of FITACF is still 2.5 and not 3.0.

**Comment by mts299** [comment]
Okay, sorry. I thought someone said 3.0 was the new default. 

Anyways, it seems to work. Not sure if anyone else wants to test it or should I merge it?

File: codebase/superdarn/src.lib/tk/oldfit.1.25/src/fitopen.c (hunk 1 above)
Comment (egthomas) [reply codebase/superdarn/src.lib/tk/oldfit.1.25/src/fitopen.c]: I hesitated to delete that line on the off chance that one day this whole git version control for the RST is abandoned and a legitimate use for that line is found.  I'm fine with it being deleted.

Comment (mts299) [reply codebase/superdarn/src.lib/tk/oldfit.1.25/src/fitopen.c]: why would it be abandoned?

Comment (egthomas) [reply codebase/superdarn/src.lib/tk/oldfit.1.25/src/fitopen.c]: Well for the first ~20 years of SD RCS was used for version control - who knows what the next fad will be?  But that's just me being silly, at this point there's no reason not to just delete the line and rely on the commit history for context.

Comment (mts299) [reply codebase/superdarn/src.lib/tk/oldfit.1.25/src/fitopen.c]: Let me know when you delete the line then we can resolve the conversation :)

**Comment by mts299** [comment]
@egthomas Shall I merge this PR as well?

**Comment by egthomas** [comment]
I think so.  It looks like your testing confirmed that the minor version numbers are now being handled correctly.

**Comment by mts299** [comment]
Merging :)

**Pull request merged**